### ISSUE TITLE
EulerAncestralDiscreteScheduler is generating weird results when used with StableDiffusionControlNetInpaintPipeline

### ISSUE THREAD

**Issue by nayan-dhabarde** [huggingface/diffusers#7348]
### Describe the bug

I was trying the demo from https://huggingface.co/docs/diffusers/en/using-diffusers/controlnet#inpainting

Only change I made was that I have used a different scheduler, i.e. instead of UniPCMultistepScheduler I have used EulerAncestralDiscreteScheduler

And the results are horrible, here is the code:
from diffusers import StableDiffusionControlNetInpaintPipeline, ControlNetModel, EulerAncestralDiscreteScheduler

controlnet = ControlNetModel.from_pretrained("lllyasviel/control_v11p_sd15_inpaint", torch_dtype=torch.float16, use_safetensors=True)
```python
pipe = StableDiffusionControlNetInpaintPipeline.from_pretrained(
    "runwayml/stable-diffusion-v1-5", controlnet=controlnet, torch_dtype=torch.float16, use_safetensors=True
)

# Setup VAE
vae = AutoencoderKL.from_pretrained("stabilityai/sd-vae-ft-mse", torch_dtype=torch.float16)
pipe.vae = vae

    # Setup scheduler
pipe.scheduler = EulerAncestralDiscreteScheduler.from_config(pipe.scheduler.config)
pipe.enable_model_cpu_offload()
```
```python
output = pipe(
    "corgi face with large ears, detailed, pixar, animated, disney",
    num_inference_steps=20,
    eta=1.0,
    image=init_image,
    mask_image=mask_image,
    control_image=control_image,
).images[0]
make_image_grid([init_image, mask_image, output], rows=1, cols=3)
```
Everything else is same as the tutorial:
Here are the results:
![image](https://github.com/huggingface/diffusers/assets/24761808/42e83a3b-fc1b-49f2-9819-93b5affbf844)
Am I missing something?

### Reproduction

from diffusers.utils import load_image, make_image_grid

init_image = load_image(
    "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/controlnet-inpaint.jpg"
)
init_image = init_image.resize((512, 512))

mask_image = load_image(
    "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/controlnet-inpaint-mask.jpg"
)
mask_image = mask_image.resize((512, 512))
make_image_grid([init_image, mask_image], rows=1, cols=2)

import numpy as np
import torch

def make_inpaint_condition(image, image_mask):
    image = np.array(image.convert("RGB")).astype(np.float32) / 255.0
    image_mask = np.array(image_mask.convert("L")).astype(np.float32) / 255.0

    assert image.shape[0:1] == image_mask.shape[0:1]
    image[image_mask > 0.5] = -1.0  # set as masked pixel
    image = np.expand_dims(image, 0).transpose(0, 3, 1, 2)
    image = torch.from_numpy(image)
    return image

control_image = make_inpaint_condition(init_image, mask_image)

from diffusers import StableDiffusionControlNetInpaintPipeline, ControlNetModel, EulerAncestralDiscreteScheduler

controlnet = ControlNetModel.from_pretrained("lllyasviel/control_v11p_sd15_inpaint", torch_dtype=torch.float16, use_safetensors=True)
pipe = StableDiffusionControlNetInpaintPipeline.from_pretrained(
    "runwayml/stable-diffusion-v1-5", controlnet=controlnet, torch_dtype=torch.float16, use_safetensors=True
)

# Setup VAE
vae = AutoencoderKL.from_pretrained("stabilityai/sd-vae-ft-mse", torch_dtype=torch.float16)
pipe.vae = vae

    # Setup scheduler
pipe.scheduler = EulerAncestralDiscreteScheduler.from_config(pipe.scheduler.config)
pipe.enable_model_cpu_offload()

output = pipe(
    "corgi face with large ears, detailed, pixar, animated, disney",
    num_inference_steps=20,
    eta=1.0,
    image=init_image,
    mask_image=mask_image,
    control_image=control_image,
).images[0]
make_image_grid([init_image, mask_image, output], rows=1, cols=3)

### System Info

RTX 3090 24GB, Runpod instance running ubuntu

### Who can help?

@sayakpaul @yiyixuxu 

### Diffusers version?
v0.27.0

**Comment by jasstionzyf** [comment]
StableDiffusionInpaintPipeline also has same issues when update to version: 2.7.0.
But previously versions like 2.6.x work normal.
Also only  not special checkpoints has problems,  inpainting models like sd1.5 inpainting has no  issues

**Comment by nayan-dhabarde** [comment]
I did checked v0.26.0 works fine. Later versions have problems

**Comment by sayakpaul** [comment]
Cc: @yiyixuxu/

With an earlier version (without changing any code), you're getting the expected results?

**Comment by nayan-dhabarde** [comment]
Yes, it is working correctly for 0.26.0, haven't tried any intermediate versions.

**Comment by standardAI** [comment]
I wonder if this resolves the issue 🤔:
```diff
-if self.begin_index is None:
+if self.begin_index is None or self.begin_index == 0:
    step_indices = [self.index_for_timestep(t, schedule_timesteps) for t in timesteps]
else:
    step_indices = [self.begin_index] * timesteps.shape[0]
```
I am not sure, I am still 🤔. The results seem proper for `EulerDiscreteScheduler` and `EulerAncestralDiscreteScheduler` rather than full noise and better for `UniPCMultistepScheduler` and `DPMSolverMultistepScheduler`.

**Comment by jasstionzyf** [comment]
still has issues compare with version 0.26.0
Previously results: 
![WechatIMG1895](https://github.com/huggingface/diffusers/assets/5241459/c2750122-f27f-49aa-b4cc-f31ae78903d9)
![WechatIMG1896](https://github.com/huggingface/diffusers/assets/5241459/499d7b25-35ba-4ac0-8aae-125afb43d820)


this fix version results:
![Uploading WechatIMG1897.jpg…]()
![Uploading WechatIMG1898.jpg…]()

**Comment by jasstionzyf** [comment]
![Uploading WechatIMG1897.jpg…]()

**Comment by yiyixuxu** [comment]
@jasstionzyf this gives me identical results between v026 and the current main. 
Would you be able to provide a script?

```python
from diffusers.utils import load_image, make_image_grid

init_image = load_image(
    "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/controlnet-inpaint.jpg"
)
init_image = init_image.resize((512, 512))

mask_image = load_image(
    "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/controlnet-inpaint-mask.jpg"
)
mask_image = mask_image.resize((512, 512))
make_image_grid([init_image, mask_image], rows=1, cols=2)

import numpy as np
import torch

def make_inpaint_condition(image, image_mask):
    image = np.array(image.convert("RGB")).astype(np.float32) / 255.0
    image_mask = np.array(image_mask.convert("L")).astype(np.float32) / 255.0

    assert image.shape[0:1] == image_mask.shape[0:1]
    image[image_mask > 0.5] = -1.0  # set as masked pixel
    image = np.expand_dims(image, 0).transpose(0, 3, 1, 2)
    image = torch.from_numpy(image)
    return image

control_image = make_inpaint_condition(init_image, mask_image)

from diffusers import StableDiffusionControlNetInpaintPipeline, ControlNetModel, EulerAncestralDiscreteScheduler, UniPCMultistepScheduler

controlnet = ControlNetModel.from_pretrained("lllyasviel/control_v11p_sd15_inpaint", torch_dtype=torch.float16, use_safetensors=True)
pipe = StableDiffusionControlNetInpaintPipeline.from_pretrained(
    "runwayml/stable-diffusion-v1-5", controlnet=controlnet, torch_dtype=torch.float16, use_safetensors=True
)

pipe.scheduler = EulerAncestralDiscreteScheduler.from_config(pipe.scheduler.config)
#pipe.scheduler = UniPCMultistepScheduler.from_config(pipe.scheduler.config)
pipe.enable_model_cpu_offload()
genrator = torch.Generator(device="cpu").manual_seed(33)

output = pipe(
    "corgi face with large ears, detailed, pixar, animated, disney",
    num_inference_steps=20,
    eta=1.0,
    image=init_image,
    mask_image=mask_image,
    generator=genrator,
    control_image=control_image,
).images[0]
make_image_grid([init_image, mask_image, output], rows=1, cols=3).save(f"yiyi_test_out_v026.png")
``` 
v026
![yiyi_test_out_v026](https://github.com/huggingface/diffusers/assets/12631849/c6d176a6-7a3d-4ed3-b17c-ae5d6d639c54)

v027
![yiyi_test_out_v027](https://github.com/huggingface/diffusers/assets/12631849/44b57e47-4c1f-49b8-b4e3-0699d6706b6c)

**Comment by jasstionzyf** [comment]
@yiyixuxu  Thanks for your patiences, my mistake, now it works as expected.

**Comment by crapthings** [comment]
I wanted invert mask, but the result turned out like this.

![4401710927696_ pic](https://github.com/huggingface/diffusers/assets/1147704/4a81ce5e-4548-4694-9865-90b0b219ef2c)

**Comment by yiyixuxu** [comment]
@crapthings 
are you specifically seeing a different result with the same code when compared with a earlier diffusers version (i.e. v0.26), or are you just not happy with the result? if it is the latter, can you open a new issue?

**Comment by crapthings** [comment]
@yiyixuxu  the result in right looks very orange

okay will test
